### PR TITLE
Migrate [internal] Userreports views to B3

### PR DESCRIPTION
**Pull request by biyeun**
@orangejenny / @benrudolph cc @czue 

Since these views are internal I'm leaning towards not doing any formal QA. I made sure each page loaded and forms submitted ok.

screen shot 2016-05-26 at 9 40 58 pm
screen shot 2016-05-26 at 9 40 48 pm
screen shot 2016-05-26 at 9 40 42 pm
screen shot 2016-05-26 at 9 40 33 pm
screen shot 2016-05-26 at 9 40 22 pm
screen shot 2016-05-26 at 9 40 13 pm
screen shot 2016-05-26 at 9 40 03 pm
screen shot 2016-05-26 at 9 39 32 pm
screen shot 2016-05-26 at 9 39 25 pm
screen shot 2016-05-26 at 9 39 13 pm

### CODE REVIEW

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -129,23 +132,105 @@ def decorated(request, domain, *args, **kwargs):
...
+            'form': self.edit_form,
+            'report': self.config,
+        }
+
+    @property
+    @memoized
+    def config(self):
+        if self.report_id is None:
+            return ReportConfiguration(domain=self.domain)
+        try:
```
Comment (czue) [inline corehq/apps/userreports/views.py]: is there a reason you changed this from using `get_report_config_or_404`?

File: corehq/apps/userreports/views.py [hunk 2]
```diff
@@ -129,23 +132,105 @@ def decorated(request, domain, *args, **kwargs):
...
+            return get_report_configs([self.report_id], self.domain)[0]
+        except ReportConfigurationNotFoundError:
+            raise Http404()
+
+    @property
+    def read_only(self):
+        return _config_id_is_static(self.report_id) if self.report_id is not None else False
```
Comment (czue) [inline corehq/apps/userreports/views.py]: should rename this function to not be "private" if you're going to import it from another module

File: corehq/apps/userreports/views.py [hunk 3]
```diff
@@ -807,40 +880,103 @@ def create_data_source_from_app(request, domain):
...
+    @property
+    def read_only(self):
+        return id_is_static(self.config_id) if self.config_id is not None else False
+
+    @property
+    @memoized
+    def config(self):
+        if self.config_id is None:
+            return DataSourceConfiguration(domain=self.domain)
+        try:
```
Comment (czue) [inline corehq/apps/userreports/views.py]: same question as above - any reason this can't use `get_datasource_config_or_404`?

**Comment by czue** [comment]
thanks for the effort on this. 

i'm ok not doing formal QA on the internal UI if you're confident in your own testing. it looks like this might have touched a couple report builder templates too - not sure how carefully you tested those but they are more important to not break now since we have lots of people in the trial

File: corehq/apps/userreports/ui/forms.py [hunk 4]
```diff
@@ -60,6 +62,37 @@ def __init__(self, domain, instance=None, read_only=False, *args, **kwargs):
...
+
+        self.helper.form_method = 'POST'
+        self.helper.form_class = 'form-horizontal'
+        self.helper.form_action = '#'
+
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-9'
+
+        self.helper.layout = crispy.Layout(
+            crispy.Fieldset(
```
Comment (czue) [inline corehq/apps/userreports/ui/forms.py]: just for my own education - why was it necessary to add this?

Comment (biyeun) [inline corehq/apps/userreports/ui/forms.py]: to add consistent styling for the `legend` text

File: corehq/apps/userreports/urls.py [hunk 5]
```diff
@@ -1,25 +1,45 @@
...
+)
 
 urlpatterns = patterns('corehq.apps.userreports.views',
-    url(r'^$', 'configurable_reports_home', name='configurable_reports_home'),
-    url(r'^reports/create/$', 'create_report', name='create_configurable_report'),
-    url(r'^reports/import/$', 'import_report', name='import_configurable_report'),
-    url(r'^reports/edit/(?P<report_id>[\w-]+)/$', 'edit_report', name='edit_configurable_report'),
+    url(r'^$', UserConfigReportsHomeView.as_view(),
+        name=UserConfigReportsHomeView.urlname
+    ),
```
Comment (dimagimon) [inline corehq/apps/userreports/urls.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/userreports/urls.py [hunk 6]
```diff
@@ -1,25 +1,45 @@
...
+    url(r'^reports/create/$', CreateConfigReportView.as_view(),
+        name=CreateConfigReportView.urlname
+    ),
```
Comment (dimagimon) [inline corehq/apps/userreports/urls.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/userreports/urls.py [hunk 7]
```diff
@@ -1,25 +1,45 @@
...
+    url(r'^reports/import/$', ImportConfigReportView.as_view(),
+        name=ImportConfigReportView.urlname
+    ),
```
Comment (dimagimon) [inline corehq/apps/userreports/urls.py]: E124 closing bracket does not match visual indentation

**Comment by biyeun** [comment]
@czue ✅

**Comment by biyeun** [comment]
@orangejenny @benrudolph ✅

**Comment by czue** [comment]
thanks - :+1: on :white_check_mark: from me. can you confirm you tested the report builder js changes or that they don't need to be tested? (feel free to merge if yes to either question and i'm offline when build passes)

**Comment by biyeun** [comment]
yep bootstrap3/main.js is the correct version that these pages should be
on, and the only changes between the two are related to css, which is all
b3 appropriate

**Comment by czue** [comment]
:boom:

**Pull request merged**